packages/cli-kit/src/public/node/notifications-system.ts

```diff
@@ -0,0 +1,235 @@
+import {versionSatisfies} from './node-package-manager.js'
+import {renderError, renderInfo, renderWarning} from './ui.js'
+import {getCurrentCommandId} from './global-context.js'
+import {fileExists, readFile} from './fs.js'
+import {outputDebug} from './output.js'
+import {CLI_KIT_VERSION} from '../common/version.js'
+import {NotificationsKey, cacheRetrieveOrRepopulate, getCache, setCache} from '../../private/node/conf-store.js'
+
+const URL = 'https://raw.githubusercontent.com/Shopify/cli/notifications/notifications.json'
+
+export interface Notifications {
+  notifications: Notification[]
+}
+
+export interface Notification {
```

Comment:
The typing here might not match the format used in the notification file -- e.g. if that file evolves but the CLI reading from it is an old one. I think it'd be helpful to defend against that -- whilst we're the authors of the notification file, we're still decoupled from it. I think I would swap this for a Zod schema, and parse the file using that. Then, if parsing fails because it doesn't match what this CLI's schema is expecting we can skip the notifications and move on. Obviously we would also try not to break the file as well.

Reply:
That's a good idea 👌

Reply:
We are catching errors [here](https://github.com/Shopify/cli/pull/4149/files#diff-0860eecbbe74005c184dce2613e954204bd54aad1b9f68806ea996dad684715cR46), so if the parsing fails, it already moves on. Any other advantage of using Zod here?